### ISSUE TITLE
myNode - unable to initialize bip84

### ISSUE THREAD

**Issue by rico1881** [Samourai-Wallet/whirlpool-gui#25]
i have setup dojo and whirlpool on myNode and both re running, but when i enter passphrase on whirlpool gui i get unable to initialize bip84 error?

i had it working before i setup dojo?

thanks

**Comment by zeroleak** [comment]
hi,
this may be a Tor problem. please try running CLI manually with:
java -jar whirlpool-client-cli-xxx-run.jar --debug --authenticate

and check for errors in the logs.

**Comment by caldon** [comment]
Hello, I've got the same problem with myNode (version 0.1.95) and Dojo, I tried running CLI manually and I got the following error (I masked my XPUB and Dojo onion address with dots):

```
2020-03-01 08:49:59.379 DEBUG 21457 --- [           main] com.samourai.http.client.JavaHttpClient  : POST.body=[apikey]
2020-03-01 08:50:04.389 DEBUG 21457 --- [           main] com.samourai.http.client.JavaHttpClient  : +GET: http://.........................................................onion/v2//fees ([Authorization])
2020-03-01 08:50:04.931 DEBUG 21457 --- [           main] c.s.w.cli.services.CliWalletService      : indexFile: whirlpool-cli-state-f7d64eb7f16e0153e76bb913aa44277d1fe7477a285b7353127b57517e356373.json
2020-03-01 08:50:04.931 DEBUG 21457 --- [           main] c.s.w.cli.services.CliWalletService      : utxosFile: whirlpool-cli-utxos-f7d64eb7f16e0153e76bb913aa44277d1fe7477a285b7353127b57517e356373.json
2020-03-01 08:50:04.939  WARN 21457 --- [           main] c.s.w.c.indexHandler.FileIndexHandler    : Unable to read /opt/mynode/whirlpool/whirlpool-cli-state-f7d64eb7f16e0153e76bb913aa44277d1fe7477a285b7353127b57517e356373.json, resetting indexes
2020-03-01 08:50:04.961  INFO 21457 --- [           main] c.samourai.wallet.client.Bip84ApiWallet  :  • Initializing bip84 wallet: 0
2020-03-01 08:50:04.974 DEBUG 21457 --- [           main] c.s.wallet.api.backend.BackendApi        : initBip84: zpub=zpub...........................................................................................................
2020-03-01 08:50:05.027 DEBUG 21457 --- [           main] c.samourai.wallet.util.oauth.OAuthImpl   : accessToken is VALID, expiresAt=Sun Mar 01 09:05:03 CST 2020
2020-03-01 08:50:05.029 DEBUG 21457 --- [           main] com.samourai.http.client.JavaHttpClient  : +POST: http://.........................................................onion/v2//xpub ([Authorization])
2020-03-01 08:50:05.030 DEBUG 21457 --- [           main] com.samourai.http.client.JavaHttpClient  : POST.body=[segwit, type, xpub]
2020-03-01 08:50:05.500 ERROR 21457 --- [           main] com.samourai.http.client.JavaHttpClient  : Http query failed: status=400, responseBody={"status":"error","error":"Unable to create new HD account"}
2020-03-01 08:50:05.523 DEBUG 21457 --- [           main] com.samourai.http.client.JavaHttpClient  : --httpClientShared
2020-03-01 08:50:05.543 ERROR 21457 --- [           main] com.samourai.http.client.JavaHttpClient  : postUrlEncoded failed: http://.........................................................onion/v2//xpub

com.samourai.wallet.api.backend.beans.HttpException: java.lang.Exception: Http query failed: status=400
        at com.samourai.http.client.JavaHttpClient.parseResponse(JavaHttpClient.java:132) ~[classes!/:na]
        at com.samourai.http.client.JavaHttpClient.postUrlEncoded(JavaHttpClient.java:103) ~[classes!/:na]
        at com.samourai.wallet.api.backend.BackendApi.initBip84(BackendApi.java:98) ~[extlibj-0.0.9.jar!/:na]
        at com.samourai.wallet.client.Bip84ApiWallet.initBip84(Bip84ApiWallet.java:111) ~[whirlpool-client-0.23.2.jar!/:na]
        at com.samourai.wallet.client.Bip84ApiWallet.<init>(Bip84ApiWallet.java:46) ~[whirlpool-client-0.23.2.jar!/:na]
        at com.samourai.wallet.client.Bip84ApiWallet.<init>(Bip84ApiWallet.java:29) ~[whirlpool-client-0.23.2.jar!/:na]
        at com.samourai.whirlpool.client.wallet.WhirlpoolWalletService.computeWhirlpoolWallet(WhirlpoolWalletService.java:150) ~[whirlpool-client-0.23.2.jar!/:na]
        at com.samourai.whirlpool.cli.services.CliWalletService.openWallet(CliWalletService.java:129) ~[classes!/:na]
        at com.samourai.whirlpool.cli.services.CliService.run(CliService.java:148) ~[classes!/:na]
        at com.samourai.whirlpool.cli.Application.run(Application.java:112) ~[classes!/:na]
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:770) ~[spring-boot-2.1.6.RELEASE.jar!/:2.1.6.RELEASE]
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:760) ~[spring-boot-2.1.6.RELEASE.jar!/:2.1.6.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:318) ~[spring-boot-2.1.6.RELEASE.jar!/:2.1.6.RELEASE]
        at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:139) ~[spring-boot-2.1.6.RELEASE.jar!/:2.1.6.RELEASE]
        at com.samourai.whirlpool.cli.Application.main(Application.java:61) ~[classes!/:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:47) ~[whirlpool.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:86) ~[whirlpool.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) ~[whirlpool.jar:na]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51) ~[whirlpool.jar:na]
Caused by: java.lang.Exception: Http query failed: status=400
        ... 23 common frames omitted
```

**Comment by zeroleak** [comment]
are you sure there is no other error preceding these lines? (even before login attempt)

**Comment by caldon** [comment]
Yes, this is the first error that appeared... Everything before these lines seemed - at least to me - correct: there were no errors and no warnings, only debug and info messages

**Comment by LaurentMT** [comment]
This sounds like an issue on Dojo side.

Can you check the error logs of the Dojo API with these commands (may require root privileges):

``` 
> cd /opt/mynode/dojo/docker/my-dojo/
> ./dojo.sh logs api -d error -n 500
```

**Comment by caldon** [comment]
Yes, in my case at least it seems like I've got the Dojo incorrectly installed - the log is full of errors of Dojo trying to access some DB tables that don't exist:

```
[20200302 03:51:51.978 040 MiB] HdAccountsService.createHdAccount()Unable to create new HD account
{ Error: ER_NO_SUCH_TABLE: Table 'samourai-main.hd' doesn't exist
    at Query.Sequence._packetToError (/home/node/app/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/home/node/app/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)
    at Protocol._parsePacket (/home/node/app/node_modules/mysql/lib/protocol/Protocol.js:278:23)
    at Parser.write (/home/node/app/node_modules/mysql/lib/protocol/Parser.js:76:12)
    at Protocol.write (/home/node/app/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/node/app/node_modules/mysql/lib/Connection.js:91:28)
    at Socket.<anonymous> (/home/node/app/node_modules/mysql/lib/Connection.js:502:10)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    --------------------
    at Pool.query (/home/node/app/node_modules/mysql/lib/Pool.js:199:23)
    at Promise (/home/node/app/lib/db/mysql-db-wrapper.js:389:19)
    at new Promise (<anonymous>)
    at MySqlDbWrapper._query (/home/node/app/lib/db/mysql-db-wrapper.js:387:12)
    at MySqlDbWrapper.ensureHDAccountId (/home/node/app/lib/db/mysql-db-wrapper.js:635:31)
    at HDAccountsService.newHdAccount (/home/node/app/lib/bitcoin/hd-accounts-service.js:133:14)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
  code: 'ER_NO_SUCH_TABLE',
  errno: 1146,
  sqlMessage: 'Table \'samourai-main.hd\' doesn\'t exist',
  sqlState: '42S02',
  index: 0,
  sql: 'SELECT `hdID` FROM `hd` WHERE `hdXpub` = \'xpub...........................................................................................................\'' }
```

**Comment by LaurentMT** [comment]
@caldon The cause of your problem is definitely there, the database used by Dojo wasn't correctly initialized during the install. If not already done, I would recommend to contact the MyNode team. They'll tell you what is the best course of action to fix the issue.

**Comment by BonXYZ** [comment]
Was there a fix for this in the end? I am now having this exact error and problem. I have reinstalled Dojo a couple times and it doesn't seem to be making a difference.

**Comment by caldon** [comment]
To me what probably helped was uninstalling dojo manually using the "dojo.sh uninstall" script:

1) I stopped Dojo in myNode GUI.
2) In terminal, I ran "/opt/mynode/dojo/docker/my-dojo/dojo.sh uninstall" command. It asked me twice for some confirmation, not sure what the 1st one was but the 2nd one was warning that "docker volume prune" command "will remove all local volumes not used by at least one container" and wether I want to continue. I confirmed both of them with "y".
3) Then I reinstalled Dojo using the myNode Settings GUI.

I don't understand Docker at all, so don't know if this is the correct solution, but at least in my case Dojo started working correctly after this.
